boards/feather_m4/examples/clocking_v2.rs

```diff
@@ -0,0 +1,171 @@
+type Uart = uart::Uart<uart::Config<Pads>, uart::Duplex>;
+
+#[app(device = atsamd_hal::pac, peripherals = true)]
+mod app {
+    use super::*;
+
+    #[shared]
+    struct SharedResources {
+        uart: Uart,
+        rtc: Rtc<ClockMode>,
+    }
+
+    #[local]
+    struct LocalResources {}
+
+    #[init]
+    fn init(cx: init::Context) -> (SharedResources, LocalResources, init::Monotonics()) {
+        let mut device = cx.device;
+
+        // Get the clocks & tokens
+        let (_buses, clocks, tokens) = clock::clock_system_at_reset(
+            device.OSCCTRL,
+            device.OSC32KCTRL,
+            device.GCLK,
+            device.MCLK,
+            &mut device.NVMCTRL,
+        );
+
+        // This is required because the `sercom` and `rtc` modules have not yet
+        // been update to use `clock::v2`
+        let (_, _, _, mut mclk) = unsafe { clocks.pac.steal() };
+
+        // Get the pins
+        let pins = Pins::new(device.PORT);
+
+        // Take `Dfll` 48 MHz, divide down to `2 MHz` through `Gclk1`
+        let (gclk1, dfll) = Gclk::from_source(tokens.gclks.gclk1, clocks.dfll);
+        let gclk1 = gclk1.div(GclkDiv16::Div(24)).enable();
+
+        // Output `Gclk1` on PB15 pin
+        let (gclk1, _gclk1_out) = gclk1.enable_gclk_out(pins.pb15);
+
+        // Setup a peripheral channel to power up `Dpll0` from `Gclk1`
+        let (pclk_dpll0, gclk1) = Pclk::enable(tokens.pclks.dpll0, gclk1);
+
+        // Configure `Dpll0` with `2 * 60 + 0/32 = 120 MHz` frequency
+        let dpll0 = Dpll::from_pclk(tokens.dpll0, pclk_dpll0)
+            .loop_div(60, 0)
+            .enable();
+
+        // Swap source of `Gclk0` from Dfll to Dpll0, `48 Mhz -> 120 MHz`
+        let (gclk0, _dfll, _dpll0) = clocks.gclk0.swap_sources(dfll, dpll0);
+
+        // Output `Gclk0` on pin PB14
+        let (gclk0, _gclk0_out) = gclk0.enable_gclk_out(pins.pb14);
+
+        // Setup a peripheral channel to power up `Dpll1` from `Gclk1`
+        let (pclk_dpll1, _gclk1) = Pclk::enable(tokens.pclks.dpll1, gclk1);
+
+        // Configure `Dpll1` with `2 * 50 + 0/32 = 100 MHz` frequency
+        let dpll1 = Dpll::from_pclk(tokens.dpll1, pclk_dpll1)
+            .loop_div(50, 0)
+            .enable();
+
+        // Output `Dpll1` on PB20 pin via `Gclk6`, divided by 200 resulting in 0.5 MHz
+        // output frequency
+        let (gclk6, _dpll1) = Gclk::from_source(tokens.gclks.gclk6, dpll1);
+        let gclk6 = gclk6.div(GclkDiv8::Div(200)).enable();
+        let (_gclk6, _gclk6_out) = gclk6.enable_gclk_out(pins.pb12);
+
+        // Configure `Xosc32k` with both outputs (1kHz, 32kHz) activated
+        let xosc32k_base = Xosc32kBase::from_crystal(tokens.xosc32k.base, pins.pa00, pins.pa01)
+            .control_gain_mode(ControlGainMode::HighSpeed)
+            .on_demand(false)
+            .run_standby(true)
+            .enable();
+        let (xosc1k, xosc32k_base) = Xosc1k::enable(tokens.xosc32k.xosc1k, xosc32k_base);
```

Comment:
Is this actually 1Khz or 1024 Hz? Might be work clarifying, although I'm not sure whether it's standard practice to call that 1KHz in industry. I suppose what we call 32khz is actually 32768, so maybe there's already a precedent in place there. Just something to consider.

Reply:
If I'm being pedantic, we only say `1k` and `32k`, which could be an abbreviation for `KHz` or `KiHz` 😆

I agree that this is slightly confusing, but I don't see a significantly better approach. Let's merge for now and if someone comes up with a better way to name things, we can change it.